### ISSUE TITLE
[spark-df] sending schema info to Cloud results in errors

### ISSUE THREAD

**Issue by vijaykiran** [sodadata/soda-core#1672]
When using schema checks with Soda Cloud enabled, scan fails with 400 response:

```
root
 |-- col_name: string (nullable = false)
 |-- data_type: string (nullable = false)
 |-- comment: string (nullable = true)
+--------------------+--------------------+-------+
|            col_name|           data_type|comment|
+--------------------+--------------------+-------+
|            CropCode|              string|       |
|             TraitID|                 int|       |
|       UtcInsertDate|           timestamp|       |
|SYSTEM_StartDateT...|           timestamp|       |
|SYSTEM_EndDateTim...|                 int|       |
|    SYSTEM_DeltaType|                 int|       |
|              __meta|struct<file_name:...|       |
|                    |                    |       |
|      # Partitioning|                    |       |
|     Not partitioned|                    |       |
+--------------------+--------------------+-------+
Error while executing Soda Cloud command response code: 400
INFO   | Soda Core 3.0.12
INFO   | Scan summary:
INFO   | 1/1 check WARNED:
INFO   |     globaltrait.croptrait in spark_df
INFO   |       Schema Check [WARNED]
INFO   |         missing_column_names = [fake]
INFO   |         schema_measured = [CropCode string, TraitID int, UtcInsertDate timestamp, SYSTEM_StartDateTimeUtc timestamp, SYSTEM_EndDateTimeUtc int, SYSTEM_DeltaType int, __meta struct<file_name:string,processing_time_utc:timestamp>,  , # Partitioning , Not partitioned ]
INFO   | Only 1 warning. 0 failure. 0 errors. 0 pass.
INFO   | Sending results to Soda Cloud
ERROR  | Error while executing Soda Cloud command response code: 400
INFO   | Soda Cloud Trace: 5295292993994965696
```

Context: https://soda-community.slack.com/archives/C038FFU79J5/p1667814395827069?thread_ts=1662967243.430199&cid=C038FFU79J5

Server Logs:
```
 "code": "invalid_request",
  "message": "Failed request validation of SodaCoreInsertScanResultsCommand on the following properties: 
 checks[0].diagnostics.schema[7].sourceDataType: must not be empty
checks[0].diagnostics.schema[7].columnName: must not be empty
checks[0].diagnostics.schema[8].sourceDataType: must not be empty
checks[0].diagnostics.schema[9].sourceDataType: must not be empty"
} (exception)
```

**Comment by jmarien** [comment]
SODA-1278

**Comment by m1n0** [comment]
@vijaykiran do you have a simple way to replicate this please? Locally I am doing 
```
...
spark_session = SparkSession.builder.master("local").appName("test").getOrCreate()
df = spark_session.createDataFrame(
    data=[{"id": "1", "name": "John Doe", "email": "valid@email.com"}],
    schema=types.StructType(
        [
            types.StructField("id", types.StringType()),
            types.StructField("name", types.StringType()),
            types.StructField("email", types.StringType()),
        ]
    ),
)
df.createOrReplaceTempView("users")

scan = Scan()
...
```
to create a testing table but this way it seems none of the meta/partitioning data is there, I get the expected output:

```
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
22/11/20 14:16:50 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
root
 |-- col_name: string (nullable = false)
 |-- data_type: string (nullable = false)
 |-- comment: string (nullable = true)

+--------+---------+-------+
|col_name|data_type|comment|
+--------+---------+-------+
|      id|   string|   null|
|    name|   string|   null|
|   email|   string|   null|
+--------+---------+-------+
INFO   | Soda Core 3.0.13
DEBUG  | Reading configuration file "sodacl_spark_df/configuration.yml"
...
INFO   | Oops! 1 failures. 0 warnings. 0 errors. 0 pass.
INFO   | Sending results to Soda Cloud
```

**Comment by vijaykiran** [comment]
Should be fixed in 3.0.16.
